src/sentry/api/endpoints/project_user_reports.py

```diff
@@ -106,9 +118,19 @@ def post(self, request, project):
             report.event_user_id = euser.id
 
         try:
-            report.group = Group.objects.from_event_id(project, report.event_id)
-        except Group.DoesNotExist:
-            pass
+            event = Event.objects.filter(project_id=project.id,
+                                         event_id=report.event_id).select_related('group')[0]
+        except IndexError:
+            try:
+                report.group = Group.objects.from_event_id(project, report.event_id)
+            except Group.DoesNotExist:
+                pass
+        else:
+            try:
+                report.environment = event.get_environment()
+            except Environment.DoesNotExist:
+                pass
```

Comment:
I think it might be worth logging if this `except Environment.DoesNotExist` is entered, since this realistically should never occur.

Reply:
@mattrobenolt just so I'm clear, this is what you're referring to? https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L740

Reply:
That's where it's created, yeah.